src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java

```diff
@@ -680,10 +680,17 @@ public String setExecutorNames(CaseData caseData,
             && caseData.getSolsWillType().matches("WillLeft")) {
             executorNames = "The executor" + returnPlural(listOfApplyingExecs) + " ";
 
-            executorNames = listOfApplyingExecs.isEmpty() ? executorNames + professionalName + ": " :
-                executorNames + FormattingService.createExecsApplyingNames(listOfApplyingExecs)
-                    +  ", " + caseData.getPrimaryApplicantForenames()
-                    + " " + caseData.getPrimaryApplicantSurname() + ": ";
+            if (caseData.getSolsSolicitorIsExec().matches(YES) && caseData.getSolsSolicitorIsApplying().matches(YES)
+                || (caseData.getTitleAndClearingType().matches("TCTTrustCorpResWithSDJ")
+                || caseData.getTitleAndClearingType().matches("TCTTrustCorpResWithApp"))) {
+                executorNames = listOfApplyingExecs.isEmpty() ? executorNames + professionalName + ": " :
+                    executorNames + FormattingService.createExecsApplyingNames(listOfApplyingExecs);
+            } else {
+                executorNames = listOfApplyingExecs.isEmpty() ? executorNames + professionalName + ": " :
+                    executorNames + FormattingService.createExecsApplyingNames(listOfApplyingExecs)
+                        + ", " + caseData.getPrimaryApplicantForenames()
+                        + " " + caseData.getPrimaryApplicantSurname() + ": ";
+            }
```

Comment:
Is it still possible to still get a null, null here? do we need null/empty checks on names or isPrmaryApplicantApplying check?

Reply:
i now check for both forename and surname being null